receiver_test.go

```diff
@@ -128,3 +130,82 @@ func TestReceiver_Nack_channel_ack_returns_error_it_should_return_the_same_error
 
 	assert.Same(t, expectedErr, err)
 }
+
+func TestReceiver_Receive(t *testing.T) {
+	dd := make(chan amqp.Delivery, 1)
+
+	now := time.Now()
+
+	dd <- amqp.Delivery{
+		Headers: amqp.Table{
+			"X-Custom-Header": []string{"test value1", "test value2"},
+		},
+		ContentType:     "test content type",
+		ContentEncoding: "test content encoding",
+		CorrelationId:   "test correlation id",
+		ReplyTo:         "test reply to",
+		Expiration:      "test expiration",
+		MessageId:       "test message id",
+		Timestamp:       now,
+		Type:            "test message type",
+		UserId:          "test user id",
+		AppId:           "test app id",
+		ConsumerTag:     "test consumer tag",
+		DeliveryTag:     123,
+	}
+
+	close(dd)
+
+	ch := &mockChannel{}
+	ch.On("Consume", "test-queue", "test-consumer", false, false, false, false, amqp.Table(nil)).Return(dd, nil)
+	ch.On("Cancel", "test-consumer", false).Return(nil)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+
+	receiver := Receiver(ch, ConsumeOptions{
+		Queue:       "test-queue",
+		ConsumerTag: "test-consumer",
+	}, "test-alias")
+
+	rch, err := receiver.Receive(ctx)
+
+	assert.Nil(t, err)
```

Comment:
I'd use `require.NoError(t, err)` for errors to make the output a bit nicer ans terminate the execution in case an error was returned, otherwise it will try to execute the next lines with a `nil` channel which will basically block forever